automation/src/test/java/org/greenplum/pxf/automation/features/json/JsonTest.java

```diff
@@ -11,198 +11,291 @@
+	 * value because the field is missing of because the array doesn't contain
+	 * the requested item.
+	 *
+	 * @throws Exception if test fails to run
+	 */
+	@Test(groups = {"features", "gpdb"})
+	public void jsonPrettyPrint() throws Exception {
+
+		exTable.setName("jsontest_pretty_print");
+		exTable.setProfile("Json");
+		exTable.setPath(hdfsPath + FILENAME_PRETTY_PRINT + SUFFIX_JSON);
+		exTable.setFields(tweetsFields);
+		exTable.setUserParameters(new String[]{"IDENTIFIER=created_at"});
+
+		gpdb.createTableAndVerify(exTable);
+
+		// Verify results
+		runTincTest("pxf.features.hdfs.readable.json.pretty_print.runTest");
+	}
+
+	/**
+	 * Test JSON file with pretty print format. Some of the records don't
+	 * contain the identifier and will be ignored.
+	 *
+	 * @throws Exception if test fails to run
+	 */
+	@Test(groups = {"features", "gpdb"})
+	public void missingIdentifier() throws Exception {
+
+		exTable.setName("jsontest_missing_identifier");
+		exTable.setProfile("Json");
+		exTable.setPath(hdfsPath + FILENAME_PRETTY_PRINT_W_DELETE + SUFFIX_JSON);
+		exTable.setFields(tweetsFields);
+		exTable.setUserParameters(new String[]{"IDENTIFIER=created_at"});
+
+		gpdb.createTableAndVerify(exTable);
+
+		// Verify results
+		runTincTest("pxf.features.hdfs.readable.json.missing_identifier.runTest");
+	}
+
+	/**
+	 * Test JSON file with pretty print format. Some of the records exceed the
+	 * max size (MAXLENGTH) and will be ignored.
+	 *
+	 * @throws Exception if test fails to run
+	 */
+	@Test(groups = {"features", "gpdb"})
+	public void exceedsMaxSize() throws Exception {
+
+		exTable.setName("jsontest_max_size");
+		exTable.setProfile("Json");
+		exTable.setPath(hdfsPath + FILENAME_PRETTY_PRINT + SUFFIX_JSON);
+		exTable.setFields(tweetsFields);
+		exTable.setUserParameters(new String[]{
+				"IDENTIFIER=created_at",
+				"MAXLENGTH=566"});
+
+		gpdb.createTableAndVerify(exTable);
+
+		// Verify results
+		runTincTest("pxf.features.hdfs.readable.json.exceed_max_size.runTest");
+	}
+
+	/**
+	 * Test JSON file with pretty print format. One of the records
+	 * is malformed. In that case the whole line will be
+	 * replaced by NULLs.
+	 *
+	 * @throws Exception if test fails to run
+	 */
+	@Test(groups = {"features", "gpdb"})
+	public void malFormatedRecord() throws Exception {
+
+		exTable.setName("jsontest_malformed_record");
+		exTable.setProfile("Json");
+		exTable.setPath(hdfsPath + FILENAME_BROKEN + SUFFIX_JSON);
+		exTable.setFields(tweetsFields);
+		exTable.setUserParameters(new String[]{"IDENTIFIER=created_at"});
+
+		gpdb.createTableAndVerify(exTable);
+
+		// Verify results
+		runTincTest("pxf.features.hdfs.readable.json.malformed_record.runTest");
+	}
+
+	/**
+	 * Test JSON file with pretty print format with reject limit configured. One of the records
+	 * is malformed. The query is allowed and a table is created.
+	 *
+	 * @throws Exception if test fails to run
+	 */
+	@Test(groups = {"features", "gpdb"})
```

Comment:
should we test this in the cloud as well?
```suggestion
	@Test(groups = {"features", "gpdb", "hcfs"})
```

Reply:
we can defer this to a different story